cupy/lib/polynomial.pyx

```diff
@@ -4,6 +4,37 @@ import cupy
 from cupy.core.core cimport ndarray
 
 
+def polyadd(a1, a2):
+    """Computes the sum of two polynomials.
+
+    Args:
+        a1 (cupy.ndarray or cupy.poly1d): first input polynomial.
+        a2 (cupy.ndarray or cupy.poly1d): second input polynomial.
+
+    Returns:
+        cupy.ndarray or cupy.poly1d: The sum of the inputs.
+
+    .. seealso:: :func:`numpy.polyadd`
+
+    """
+    truepoly = False
+    if isinstance(a1, poly1d):
+        a1 = a1.coeffs
+        truepoly = True
+    if isinstance(a2, poly1d):
+        a2 = a2.coeffs
+        truepoly = True
+    a1 = cupy.atleast_1d(a1)
+    a2 = cupy.atleast_1d(a2)
+    if a1.shape[0] < a2.shape[0]:
+        a1, a2 = a2, a1
+    a2 = cupy.pad(a2, (a1.shape[0] - a2.shape[0], 0))
+    val = a1 + a2
+    if truepoly:
+        val = poly1d(val)
```

Comment:
`trimseq` in `poly1d` instantiation requires device synchronization. So, using `polyadd` for `poly1d` inputs is much slower than for `ndarray`. Why not implement lazy evaluation for `coeffs` and `order`?

Reply:
Can you elaborate a bit?

Reply:
Could you split the PR for this fix?

Reply:
Please fix not to avoid synchronize devices.

Reply:
Please elaborate on your comment. I don't think I get your point.

Reply:
Please fix to access `_coeffs` and fix tests to check `polyadd` does not synchronize devices.